.github/workflows/update_dependencies.yml

```diff
@@ -0,0 +1,56 @@
+name: Update depencies in poetry lockfile
+
+on:
+  schedule:
+    - cron:  '0 1 * * 1'
+  workflow_dispatch:
+
+jobs:
+  update_dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      # Check out repo and set up Python
+      - uses: actions/checkout@v3
+        with:
+          lfs: true
+
+      - uses: tibdex/github-app-token@v1
+        id: generate-token
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+
+      # Use cached python and dependencies, install poetry
+      - name: "Setup Python, Poetry and Dependencies"
+        uses: packetcoders/action-setup-cache-python-poetry@main
+        with:
+          python-version: 3.8
+          poetry-version: 1.2.2
+
+      # update poetry lockfile
+      - name: "Update poetry lock file"
+        id: update
+        run: |
+          poetry run poetry self update
```

Comment:
I dont think a poetry run is needed. poetry self update should work fine.

Reply:
We need to do this in the context of the poetry environment, no? Maybe its not needed, but it works

Reply:
You are right, it's not needed